lightning/src/util/enforcing_trait_impls.rs

```diff
@@ -27,6 +27,27 @@ impl EnforcingChannelKeys {
 		}
 	}
 }
+
+impl EnforcingChannelKeys {
+	fn check_keys<T: secp256k1::Signing + secp256k1::Verification>(&self, secp_ctx: &Secp256k1<T>,
+	                                                               keys: &TxCreationKeys) -> Result<(), ()> {
+		let revocation_base = PublicKey::from_secret_key(secp_ctx, &self.inner.revocation_base_key);
+		let payment_base = PublicKey::from_secret_key(secp_ctx, &self.inner.payment_base_key);
+		let htlc_base = PublicKey::from_secret_key(secp_ctx, &self.inner.htlc_base_key);
+
+		let remote_points = self.inner.remote_channel_pubkeys.as_ref().unwrap();
+
+		let keys_expected = TxCreationKeys::new(secp_ctx,
+		                                        &keys.per_commitment_point,
+		                                        &remote_points.delayed_payment_basepoint,
+		                                        &remote_points.htlc_basepoint,
+		                                        &revocation_base,
+		                                        &payment_base,
+		                                        &htlc_base).unwrap();
+		if keys == &keys_expected { Ok(()) } else { Err(()) }
```

Comment:
This should just panic!() - the failure mode for EnforcingChannelKeys where the callee (ie somewhere else in rust-lightning) made a mistake should be panic, not "safe" Err return.

Reply:
done